### PR TITLE
bsp: Add ESP32-S3-Korvo-2 Board Support Package + audio example

### PR DESCRIPTION
**Pull request by espzav**
# Checklist for new Board Support package or Component

- [x] Component contains License
- [x] Component contains README.md
- [x] Project [README.md](../README.md) updated
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to CI [upload job](https://github.com/espressif/esp-bsp/blob/master/.github/workflows/upload_component.yml#L17)
- [ ] New files were added to CI build job
- [ ] _Optional:_ Component contains unit tests
- [ ] CI passing

# Description
Added new board ESP32-S3-Korvo-2 with simple `audio` example, which was copied from `display_audio` and removed display and RGB features. 

**Tested on HW:**
- Audio output (wav to speaker)
- Audio input (record to wav and play)
- Both LEDs
- Buttons
- SD card (read wav and play)
- Camera (only init)

**Missing features:**
- Display (missing HW)
- Battery measurement (need to be updated IOT button component - more https://github.com/espressif/esp-iot-solution/issues/234)

**Audio example added:**
I added new audio example. I wanted to use the `display_audio` with some `#ifdef` but there would be lot of conditions for display, removing RGB, select audio input (es8311 / es7210) and one button name is different. I decided make new example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +27,7 @@ Best way to start with ESP-BSP is trying one of the [examples](examples) on your
 
 | Example | Supported boards |
 |---|---|
+| [audio](examples/audio) | ESP32-S3-Korvo-2 |
```
Comment (tore-espressif) [inline README.md:30]: Not related to this PR, but could you remove the `rainmaker_example` from the list? We removed it from esp-bsp a while ago

File: esp32_s3_korvo_2/include/bsp/esp32_s3_korvo_2.h [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+#include "driver/i2c.h"
+#include "driver/i2s_std.h"
+#include "soc/usb_pins.h"
+#include "driver/sdmmc_host.h"
+#include "iot_button.h"
+#include "esp_io_expander.h"
+#include "lvgl.h"
+
+/**************************************************************************************************
+ *  ESP-BOX pinout
```
Comment (tore-espressif) [inline esp32_s3_korvo_2/include/bsp/esp32_s3_korvo_2.h:20]: ESP-BOX referenced in several places

File: esp32_s3_korvo_2/esp32_s3_korvo_2.c [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+        .pull_down_en = GPIO_PULLDOWN_DISABLE,
+        .intr_type = GPIO_INTR_DISABLE
+    };
+    return gpio_config(&button_io_config);
+}
+
+bool bsp_button_get(const bsp_button_t btn)
+{
+    return !(bool)gpio_get_level(btn);
+}
```
Comment (tore-espressif) [inline esp32_s3_korvo_2/esp32_s3_korvo_2.c:535]: Is this even working? Have a look at the Korvo schematic: https://dl.espressif.com/dl/schematics/SCH_ESP32-S3-KORVO-2_V3_0_20210918.pdf

All buttons share one GPIO. You must measure voltage on the pin. These two functions should use `iot_button` API

File: esp32_s3_korvo_2/esp32_s3_korvo_2.c [hunk 4]
```diff
@@ -0,0 +1,596 @@
...
+    BSP_ERROR_CHECK_RETURN_NULL(lvgl_port_tick_init());
+    lvgl_mux = xSemaphoreCreateMutex();
+    BSP_NULL_CHECK(lvgl_mux, NULL);
+    xTaskCreate(lvgl_port_task, "LVGL task", 4096, NULL, CONFIG_BSP_DISPLAY_LVGL_TASK_PRIORITY, NULL);
+    return disp;
+#else
+    ESP_LOGW(TAG, "Display is disabled, because it is not tested yet!");
+    return NULL;
+#endif
+}
```
Comment (tore-espressif) [inline esp32_s3_korvo_2/esp32_s3_korvo_2.c:489]: Could we use the new `esp_lvgl_port` component?

File: esp32_s3_korvo_2/esp32_s3_korvo_2.c [hunk 5]
```diff
@@ -0,0 +1,596 @@
...
+        .pull_up_en = GPIO_PULLDOWN_DISABLE,
+    };
+    BSP_ERROR_CHECK_RETURN_ERR(gpio_config(&io_conf));
+
+    return ESP_OK;
+}
+
+esp_err_t bsp_audio_poweramp_enable(bool enable)
+{
+    BSP_ERROR_CHECK_RETURN_ERR(gpio_set_level(BSP_POWER_AMP_IO, enable ? 1 : 0));
```
Comment (tore-espressif) [inline esp32_s3_korvo_2/esp32_s3_korvo_2.c:190]: I can see from the schematic the this signal is connected through the IO expander. Does `gpio_set_level` work here?

**Review by tore-espressif — COMMENTED** [review]
@espzav Only a partial review for now. I'm going to check on HW

File: esp32_s3_korvo_2/esp32_s3_korvo_2.c (hunk 4 above)
Comment (espzav) [reply esp32_s3_korvo_2/esp32_s3_korvo_2.c:489]: Yes, but we haven't got HW for the LCD. I prepared before the component was released and disabled it. I didn't want to remove it. Is it ok to change it to lvgl_port, after we will have the LCD HW?

File: esp32_s3_korvo_2/esp32_s3_korvo_2.c (hunk 5 above)
Comment (espzav) [reply esp32_s3_korvo_2/esp32_s3_korvo_2.c:190]: I thought same, that it is only on IO expander. But I found, that it is connected to ESP GPIO pin. It must be controlled via this pin.

File: esp32_s3_korvo_2/esp32_s3_korvo_2.c (hunk 3 above)
Comment (espzav) [reply esp32_s3_korvo_2/esp32_s3_korvo_2.c:535]: I know, it is in example. I forgot remove this part of code. Thanks for note.

**Comment by espzav** [comment]
@tore-espressif Thank you for the review. Fixed some issues, others in comments.

File: examples/audio/main/bsp_audio_example.c [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+    /* Microphone settings */
+    es7210_dev_handle_t es7210_handle = NULL;
+    const es7210_i2c_config_t es7210_i2c = BSP_ES7210_I2C_CONFIG(BSP_I2C_NUM);
+    ESP_ERROR_CHECK(es7210_new_codec(&es7210_i2c, &es7210_handle));
+    const es7210_codec_config_t codec_conf = BSP_ES7210_CONFIG(22050, I2S_MCLK_MULTIPLE_512);
+    ESP_ERROR_CHECK(es7210_config_codec(es7210_handle, &codec_conf));
+    ESP_ERROR_CHECK(es7210_config_volume(es7210_handle, 32));
+
+    /* Configure I2S peripheral and Power Amplifier */
+    bsp_audio_init(NULL, &i2s_tx_chan, &i2s_rx_chan);
```
Comment (tore-espressif) [inline examples/audio/main/bsp_audio_example.c:118]: This function sets-up I2S for stereo operation, but the .wav file is mono. So it plays twice as fast on my setup

**Review by tore-espressif — APPROVED** [review]
LGTM, just one comment about audio playback.

File: examples/audio/main/bsp_audio_example.c (hunk 6 above)
Comment (espzav) [reply examples/audio/main/bsp_audio_example.c:118]: We will solve in next version (I hope).

**Pull request merged**